File: scripts/DesignPatterns/Singleton/Dice.php
<?php

/**
 * Dice Class
 * @author Jose Guillermo <jguillermo@outlook.com>
 * @copyright (c) 2017, Orbis
 */
class Dice
{
    private static $instance;
    /**
     * Dice constructor.
     */
    private function __construct()
    {
    }

    /**
     * @return Dice
     */
    public static function getInstance()
    {
        if(self::$instance == null){
            echo "Se creo la instancia solo una vez";
            self::$instance = new Dice();
        }
        return self::$instance;
    }

    public function roll()
    {
        return rand(1, 6);
    }
}

File: scripts/DesignPatterns/Singleton/Dice.java
package DesignPatterns.Singleton;

public class Dice {

    private static Dice instance = null;

    private Dice(){

    }

    public static Dice getInstance() {
        if (instance == null){
            instance = new Dice();
        }
        return instance;
    }

    public int roll(){
        return (int) (Math.random() * 5) + 1;
    }
}
